File: Two Pointers/valid-parentheses.java
public class Solution {
    public boolean isValid(String s) {

        if (s.length() % 2 != 0) {
            return false;
        }

        while (s.contains("()") || s.contains("{}") || s.contains("[]")) {
            System.out.println(s);
            s = s.replace("()", "");
            System.out.println(s);
            s = s.replace("{}", "");
            System.out.println(s);
            s = s.replace("[]", "");
            System.out.println(s);
        }

        return s.isEmpty();
    }
}

File: Stack/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        # while "()" in s or "[]" in s or "{}" in s:
        #     s = s.replace("()", "")
        #     s = s.replace("[]", "")
        #     s = s.replace("{}", "")
        # return s == ""

        # stack approach
        stack = []
        for char in s:
            if char in ["(", "[", "{"]:

                stack.append(char)
            elif char in [")", "]", "}"]:
                if not stack:
                    return False
                top = stack.pop()
                if char == ")" and top != "(":
                    return False
                if char == "]" and top != "[":
                    return False
                if char == "}" and top != "{":
                    return False
        return not stack
